mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -362,14 +362,20 @@ mlir::scf::tileUsingSCFForOp(RewriterBase &rewriter, TilingInterface op,
   auto clonedOp = cast<TilingInterface>(
       cloneOpAndUpdateDestinationArgs(rewriter, op, clonedOpDestination));
 
-  // 5b. Tile the cloned operation.
+  // 5b. Early return cloned op if tiling is not happening.
```

Comment:
This might work, but maybe we should just not clone the op if tile sizes are zero.

Reply:
I actually tried that, but it crashes about rewriter states. I did not dig into details because I have less experience about that...

Reply:
My take is that there are issues about replace op with itself. Do you want me to look into the crash further or add more comments to the code and commit?

Reply:
I just tested it locally, and think it makes sense to return the cloned op. The suggestion will lead to `rewriter.replace(op, op->getResults())` and get the crash. Returning the cloned op instead of `op` itself or `failure` makes more sense in this case. I will add some comments to it.

Reply:
Yeah, can't replace an op with itself. In theory you could say the caller should check for this and not do the replace, but that is strange. Let's go with this.